conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,49 @@
+import os
+from conan.api.output import Color
+from conan.tools.env import VirtualBuildEnv, Environment
+from conan.tools.files import save
+
+
+class ROSEnv:
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.sh file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._variables = {}
+        self.variables = {}
+        self._virtualbuildenv = VirtualBuildEnv(self._conanfile, auto_generate=True)
+        self._virtualbuildenv.basename = "conanrosenv"
+        self._rosenv_wrapper = "conanrosenv.sh"
+
+    def generate(self):
+        output_folder = self._conanfile.generators_folder
+        self._variables["CMAKE_TOOLCHAIN_FILE"] = os.path.join(output_folder, "conan_toolchain.cmake")
+        build_type = self._conanfile.settings.get_safe("build_type")
+        if build_type:
+            self._variables["CMAKE_BUILD_TYPE"] = build_type
+        self.variables.update(self._variables)
+
+        # Add ROS required variables to VirtualBuildEnv
+        rosbuildenv = Environment()
+        for k, v in self.variables.items():
+            rosbuildenv.define(k, v)
+        self._virtualbuildenv._buildenv = rosbuildenv
```

Comment:
I'd say that we don't need to mess, not even instantiate the ``VirtualBuildEnv`` at all